app/services/create_vote_service.rb

```diff
@@ -0,0 +1,50 @@
+class CreateVoteService
+
+  def initialize(votable,current_user,params)
+    @votable = votable
+    @current_user = current_user
+    @params = params
+  end
+
+  def call
+    if @votable.votes.where(user_id: @current_user.id).present?
+      vote = @votable.votes.where(user_id: @current_user.id).first
```

Comment:
Instead of defining a variable here, I would suggesting making an `existing_vote` private method below that memoizes this query. That could simplify your first if statement as well to `if existing_vote.present?`